frontend/src/metabase/actions/containers/ActionCreator/FormCreator/FieldSettingsPopover.tsx

```diff
@@ -189,19 +189,19 @@ function RequiredInput({
         <RequiredToggleLabel htmlFor="is-required">{t`Required`}</RequiredToggleLabel>
         <Toggle
           id="is-required"
-          value={!!value}
-          onChange={required => onChange({ required, defaultValue })}
+          value={value}
+          onChange={required => onChange({ required })}
```

Comment:
Use the same pattern we use in QB - drop the default value when `required` flag is changed. This removes a lot of weird issues we have to deal with, like setting the default value only for required fields.

Reply:
It doesn't seem like we cover the `defaultValue` reset in unit tests; mind adding a test for it / changing an existing one?

Reply:
Modified the existing test for that